src/Microsoft.DotNet.Darc/DarcLib/LocalGitClient.cs

```diff
@@ -443,6 +442,22 @@ public async Task<string> BlameLineAsync(string repoPath, string relativeFilePat
         return result.StandardOutput.Trim().Split(' ').First();
     }
 
+    public async Task<bool> GitRefExists(string repoPath, string gitRef, CancellationToken cancellationToken = default)
```

Comment:
is this needed? Aren't we making it so you can only flow builds, and we know the commit of a build

Reply:
The bottom part, might never get used, if we're always passing commit shas

Reply:
The SHA detection was actually already there and working before this change. This change specifically makes it work with branches which we need to fetch (like the target PR branch). We need to fetch/verify if a branch exists in any remote